helix-loader/src/grammar.rs

```diff
@@ -89,15 +89,106 @@ pub fn fetch_grammars() -> Result<()> {
     let mut grammars = get_grammar_configs()?;
     grammars.retain(|grammar| !matches!(grammar.source, GrammarSource::Local { .. }));
 
-    run_parallel(grammars, fetch_grammar, "fetch")
+    print!("Fetching {} grammars", grammars.len());
+    let results = run_parallel(grammars, fetch_grammar);
+    println!(" ... done");
+
+    let mut errors = Vec::new();
+    let mut git_updated = Vec::new();
+    let mut git_up_to_date = Vec::new();
+    let mut non_git = Vec::new();
+
+    for res in results {
+        match res {
+            Ok(FetchSuccess {
+                grammar_id,
+                action: FetchAction::GitUpdated { revision },
+            }) => git_updated.push(format!("{}->{}", grammar_id, revision)),
+            Ok(FetchSuccess {
+                grammar_id,
+                action: FetchAction::GitUpToDate,
+            }) => git_up_to_date.push(grammar_id),
+            Ok(FetchSuccess {
+                grammar_id,
+                action: FetchAction::NonGit,
+            }) => non_git.push(grammar_id),
+            Err(e) => errors.push(e),
+        }
+    }
+
+    // People will easily find the grammar they want that way
+    git_updated.sort_unstable();
+    git_up_to_date.sort_unstable();
+    non_git.sort_unstable();
+
+    if !git_up_to_date.is_empty() {
+        println!("{:02} up to date git grammars", git_up_to_date.len());
```

Comment:
I think I prefer it to not have leading zeros. I don't think alignment of the digits is necessary

Reply:
Done